dali/pipeline/data/tensor_list.h

```diff
@@ -342,30 +320,19 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
 
   DLL_PUBLIC inline TensorList<Backend>& operator=(TensorList<Backend> &&other) noexcept {
     if (&other != this) {
-      // Steal all data and set input to default state
-      data_ = other.data_;
       shape_ = std::move(other.shape_);
-      size_ = other.size_;
       offsets_ = std::move(other.offsets_);
-      type_ = other.type_;
-      num_bytes_ = other.num_bytes_;
-      device_ = other.device_;
       tensor_views_ = std::move(other.tensor_views_);
-      shares_data_ = other.shares_data_;
-      pinned_ = other.pinned_;
       meta_ = std::move(other.meta_);
       layout_ = std::move(other.layout_);
 
-      other.data_ = nullptr;
       other.shape_ = {};
-      other.size_ = 0;
-      other.offsets_ = {};
-      other.type_ = TypeInfo::Create<NoType>();
-      other.num_bytes_ = 0;
       other.tensor_views_.clear();
-      other.shares_data_ = false;
+      other.offsets_.clear();
       other.meta_ = {};
```

Comment:
Either both `clear()` or both `= {}`. They are both vectors

Reply:
Hmm.. `meta_ = {}` was taken from `Tensor` where it's not a vector. I can change it to clear.

Reply:
just for the sake of consistency